keps/sig-api-machinery/3157-watch-list/README.md

```diff
@@ -852,21 +868,157 @@ will rollout across nodes.
+```
+
+Check if informers used by the `kcm` fell back to standard LIST/WATCH semantics.
+```
+kubectl logs -n kube-system kube-controller-manager-kind-control-plane | grep -e "watch-list"
+W1002 09:11:40.656641       1 reflector.go:340] The watch-list feature is not supported by the server, falling back to the previous LIST/WATCH semantics
+…
+```
+
+Disable the `WatchList` feature gate for the `kcm` by editing the static pod manifest directly.
+```
+docker exec -ti kind-control-plane bash
+vim /etc/kubernetes/manifests/kube-controller-manager.yaml
+```
+and pass `- --feature-gates=WatchList=false` to the `kcm` container.
+
+Check if `kcm` is running.
+```
+kubectl get po -n kube-system
+NAME                                         READY   STATUS    RESTARTS        AGE
+…
+kube-controller-manager-kind-control-plane   1/1     Running   0               12s
+```
+
+Check if the `kas` has not recorded the watchlist latency metric.
+```
+kubectl get --raw '/metrics' | grep "apiserver_watch_list_duration_seconds"
+```
+
+Check if there are no traces of informers for `kcm` falling back to standard LIST/WATCH semantics.
+```
+kubectl logs -n kube-system kube-controller-manager-kind-control-plane | grep -e "watch-list"
+```
+
+Enable the `WatchList` feature gate for the `kas` by editing the static pod manifest directly.
+```
+docker exec -ti kind-control-plane bash
+vim /etc/kubernetes/manifests/kube-apiserver.yaml 
+```
+and remove `- --feature-gates=WatchList=false` from the `kas` container.
+
+Check if `kcm` is running.
+```
+kubectl get po -n kube-system
+NAME                                         READY   STATUS             RESTARTS      AGE
+…
+kube-controller-manager-kind-control-plane   1/1     Running            1 (22s ago)   86s
+```
+
+Check if the `kas` has not recorded the watchlist latency metric.
+```
+kubectl get --raw '/metrics' | grep "apiserver_watch_list_duration_seconds"
+```
+
+Enable the `WatchList` feature gate for the `kcm` by editing the static pod manifest directly.
+```
+docker exec -ti kind-control-plane bash
+vim /etc/kubernetes/manifests/kube-controller-manager.yaml
+```
+and remove `- --feature-gates=WatchList=false` for the `cm` container.
 
+Check if `kcm` is running.
+```
+kubectl get po -n kube-system
+NAME                                         READY   STATUS    RESTARTS      AGE
+…
+kube-controller-manager-kind-control-plane   1/1     Running   0             13s
+```
+
+Check if the `kas` has recorded the watchlist latency metric.
+```
+kubectl get --raw '/metrics' | grep "apiserver_watch_list_duration_seconds"
+# HELP apiserver_watch_list_duration_seconds [ALPHA] Response latency distribution in seconds for watch list requests broken by group, version, resource and scope.
+# TYPE apiserver_watch_list_duration_seconds histogram
+…
+apiserver_watch_list_duration_seconds_bucket{group="",resource="configmaps",scope="cluster",version="v1",le="6"} 1
+```
```

Comment:
This is great test overall

I'm fine with it and we should run it right before promoting FG to beta [and it's fine if we will need to adjust something at this point]